File: sdmt-usecase/src/main/scala/dev/tchiba/sdmt/usecase/domainmodel/create/CreateDomainModelInput.scala
package dev.tchiba.sdmt.usecase.domainmodel.create

import dev.tchiba.arch.usecase.Input
import dev.tchiba.sdmt.core.boundedContext.BoundedContextId
import dev.tchiba.sdmt.core.domainmodel.{EnglishName, UbiquitousName, Knowledge}

final case class CreateDomainModelInput(
    boundedContextId: BoundedContextId,
    ubiquitousName: UbiquitousName,
    englishName: EnglishName,
    knowledge: Knowledge
) extends Input[CreateDomainModelOutput]

File: front/angular/src/app/models/domainModel/state/io/create-domain-model-input.ts
import { BoundedContextId } from '../../../boundedContext/id/bounded-context-id';
import { UbiquitousName } from '../../ubiquitousName/ubiquitous-name';
import { EnglishName } from '../../englishName/english-name';
import { Knowledge } from '../../knowledge/knowledge';

export interface CreateDomainModelInput {
  boundedContextId: BoundedContextId;
  ubiquitousName: UbiquitousName;
  englishName: EnglishName;
  knowledge: Knowledge;
}
